addons/web/static/tests/core/popover/popover_tests.js

```diff
@@ -159,3 +159,37 @@ QUnit.test("reposition popover should properly change classNames", async (assert
     );
     assert.strictEqual(arrow.className, "popover-arrow top-auto");
 });
+
+QUnit.test("within iframe", async (assert) => {
+    const iframe = document.createElement("iframe");
+    iframe.srcdoc = `<div id="target">Within iframe</div>`;
+    const def = makeDeferred();
+    iframe.onload = def.resolve;
+    fixture.appendChild(iframe);
+    await def;
+
+    let popoverEl;
+    const TestPopover = class extends Popover {
+        onPositioned(el, { direction }) {
+            popoverEl = el;
+            assert.step(direction);
+        }
+    };
+
+    popoverTarget = iframe.contentDocument.getElementById("target");
+    await mount(TestPopover, fixture, {
+        props: { target: popoverTarget, iframe },
+    });
+    assert.verifySteps(["bottom"]);
+
+    // The popover should be rendered outside the iframe
+    assert.containsOnce(fixture, ".o_popover");
+    assert.containsNone(iframe.contentDocument.documentElement, ".o_popover");
+
+    // The popover should be rendered in the correct position
+    const { top, left, width } = popoverEl.getBoundingClientRect();
+    const { top: targetTop, left: targetLeft } = popoverTarget.getBoundingClientRect();
+    const { top: iframeTop, left: iframeLeft } = iframe.getBoundingClientRect();
+    assert.strictEqual(top, iframeTop + targetTop + popoverTarget.offsetHeight);
+    assert.strictEqual(left, iframeLeft + targetLeft + popoverTarget.offsetWidth / 2 - width / 2);
+});
```

Comment:
This test should also assert than after a scroll event on the iframe element, the popover position is updated.